akka-remote-tests/src/multi-jvm/scala/akka/remote/RemoteRestartedQuarantinedSpec.scala

```diff
@@ -0,0 +1,139 @@
+    def receive = {
+      case "shutdown" ⇒ context.system.shutdown()
+      case "identify" ⇒ sender() ! (AddressUidExtension(context.system).addressUid, self)
+    }
+  }
+
+}
+
+class RemoteRestartedQuarantinedSpecMultiJvmNode1 extends RemoteRestartedQuarantinedSpec
+class RemoteRestartedQuarantinedSpecMultiJvmNode2 extends RemoteRestartedQuarantinedSpec
+
+abstract class RemoteRestartedQuarantinedSpec
+  extends MultiNodeSpec(RemoteRestartedQuarantinedSpec)
+  with STMultiNodeSpec with ImplicitSender {
+
+  import RemoteRestartedQuarantinedSpec._
+
+  override def initialParticipants = 2
+
+  def identifyWithUid(role: RoleName, actorName: String): (Int, ActorRef) = {
+    system.actorSelection(node(role) / "user" / actorName) ! "identify"
+    expectMsgType[(Int, ActorRef)]
+  }
+
+  "A restarted quarantined system" must {
+
+    "should not crash the other system (#17213)" taggedAs LongRunningTest in {
+
+      system.actorOf(Props[Subject], "subject")
+      enterBarrier("subject-started")
+
+      runOn(first) {
+        val secondAddress = node(second).address
+
+        val (uid, ref) = identifyWithUid(second, "subject")
+
+        RARP(system).provider.transport.quarantine(node(second).address, Some(uid))
+
+        enterBarrier("quarantined")
+        enterBarrier("still-quarantined")
+
+        testConductor.shutdown(second).await
+
+        within(30.seconds) {
+          awaitAssert {
+            system.actorSelection(RootActorPath(secondAddress) / "user" / "subject") ! Identify("subject")
+            expectMsgType[ActorIdentity](1.second).ref.get
+          }
+        }
+
+        system.actorSelection(RootActorPath(secondAddress) / "user" / "subject") ! "shutdown"
+      }
+
+      runOn(second) {
+        val addr = system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val firstAddress = node(first).address
+
+        val (_, ref) = identifyWithUid(first, "subject")
+
+        enterBarrier("quarantined")
+
+        // Check that quarantine is intact
+        within(10.seconds) {
+          awaitAssert {
+            EventFilter.warning(pattern = "The remote system has quarantined this system", occurrences = 1).intercept {
+              ref ! "boo!"
+            }
+          }
+        }
+
+        enterBarrier("still-quarantined")
+
+        system.awaitTermination(10.seconds)
+
+        val freshSystem = ActorSystem(system.name, ConfigFactory.parseString(s"""
+                    akka.remote.retry-gate-closed-for = 0.5 s
+                    akka.remote.netty.tcp {
+                      hostname = ${addr.host.get}
+                      port = ${addr.port.get}
+                    }
+                    """).withFallback(system.settings.config))
+
+        val probe = TestProbe()(freshSystem)
+
+        freshSystem.actorSelection(RootActorPath(firstAddress) / "user" / "subject").tell(Identify("subject"), probe.ref)
+        probe.expectMsgType[ActorIdentity](1.second).ref.get
```

Comment:
Me neither, but that has nothing to do with the fix here since it is just a
removal of an assertion. This means I have to investigate this second bug,
too.
2015.04.22. 20:06 ezt írta ("Patrik Nordwall" <notifications@github.com>):

> In
> akka-remote-tests/src/multi-jvm/scala/akka/remote/RemoteRestartedQuarantinedSpec.scala
> <https://github.com/akka/akka/pull/17263#discussion_r28898917>:
>
> > +        enterBarrier("still-quarantined")
> > +
> > +        system.awaitTermination(10.seconds)
> > +
> > +        val freshSystem = ActorSystem(system.name, ConfigFactory.parseString(s"""
> > +                    akka.remote.retry-gate-closed-for = 0.5 s
> > +                    akka.remote.netty.tcp {
> > +                      hostname = ${addr.host.get}
> > +                      port = ${addr.port.get}
> > +                    }
> > +                    """).withFallback(system.settings.config))
> > +
> > +        val probe = TestProbe()(freshSystem)
> > +
> > +        freshSystem.actorSelection(RootActorPath(firstAddress) / "user" / "subject").tell(Identify("subject"), probe.ref)
> > +        probe.expectMsgType[ActorIdentity](1.second).ref.get
>
> yes, that is what I'm using, but with 20 sec.
> I can't get it to fail with with 30 sec.
>
> I still need to understand why the first message is not getting through.
> Is the old quarantine blocking the fresh system, or is it gated?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/17263/files#r28898917>.
>